test/micro/org/openjdk/bench/jdk/incubator/vector/RotateBenchmark.java

```diff
@@ -0,0 +1,210 @@
+// 2 along with this work; if not, write to the Free Software Foundation,
+// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+// or visit www.oracle.com if you need additional information or have any
+// questions.
+//
+//
+package org.openjdk.bench.jdk.incubator.vector;
+
+import java.util.Random;
+import jdk.incubator.vector.*;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Thread)
+public class RotateBenchmark {
+    @Param({"64","128","256"})
+    int size;
+
+    @Param({"128","256", "512"})
+    int bits;
+
+    @Param({"7","15","31"})
+    int shift;
+
+    byte[] byteinp;
+    byte[] byteres;
+    short[] shortinp;
+    short[] shortres;
+    int[] intinp;
+    int[] intres;
+    long[] longinp;
+    long[] longres;
+    VectorSpecies bspecies;
+    VectorSpecies sspecies;
+    VectorSpecies ispecies;
+    VectorSpecies lspecies;
+
+    static final byte[] specialvalsbyte = {0, -0, Byte.MIN_VALUE, Byte.MAX_VALUE};
+    static final short[] specialvalsshort = {0, -0, Short.MIN_VALUE, Short.MAX_VALUE};
+    static final int[] specialvalsint = {0, -0, Integer.MIN_VALUE, Integer.MAX_VALUE};
+    static final long[] specialvalslong = {0L, -0L, Long.MIN_VALUE, Long.MAX_VALUE};
+
+    @Setup(Level.Trial)
+    public void BmSetup() {
+        Random r = new Random(1024);
+        byteinp = new byte[size];
+        byteres = new byte[size];
+        shortinp = new short[size];
+        shortres = new short[size];
+        intinp = new int[size];
+        intres = new int[size];
+        longinp = new long[size];
+        longres = new long[size];
+
+        bspecies = VectorSpecies.of(byte.class, VectorShape.forBitSize(bits));
+        sspecies = VectorSpecies.of(short.class, VectorShape.forBitSize(bits));
+        ispecies = VectorSpecies.of(int.class, VectorShape.forBitSize(bits));
+        lspecies = VectorSpecies.of(long.class, VectorShape.forBitSize(bits));
+
+        for (int i = 4; i < size; i++) {
+            byteinp[i] = (byte)i;
+            shortinp[i] = (short)i;
+            intinp[i] = i;
+            longinp[i] = i;
+        }
+        for (int i = 0 ; i < specialvalsbyte.length; i++) {
+            byteinp[i] = specialvalsbyte[i];
+        }
+        for (int i = 0 ; i < specialvalsshort.length; i++) {
+            shortinp[i] = specialvalsshort[i];
+        }
+        for (int i = 0 ; i < specialvalsint.length; i++) {
+            intinp[i] = specialvalsint[i];
+        }
+        for (int i = 0 ; i < specialvalslong.length; i++) {
+            longinp[i] = specialvalslong[i];
+        }
+    }
+
+    @Benchmark
+    public void testRotateLeftB(Blackhole bh) {
+        ByteVector bytevec = null;
+        for (int j = 0 ; j < size; j+= bspecies.length()) {
```

Comment:
```suggestion
        for (int j = 0 ; j < size; j += bspecies.length()) {
```
Needs a space between `+=`.